src/cpp/rtps/builtin/discovery/endpoint/EDPServer.cpp

```diff
@@ -161,9 +177,9 @@ bool EDPServer::trimWriterHistory(
     // sweep away any resurrected endpoint
     for (auto iD = mp_PDP->ParticipantProxiesBegin(); iD != mp_PDP->ParticipantProxiesEnd(); ++iD)
     {
-        ProxyCont& readers = (*iD)->*pC;
+        ProxyCont& readers = *(*iD->*pC);
 
-        for (auto iE : *readers)
+        for (auto iE : readers)
```

Comment:
Please, rename the variable to a more generic name, like `endpoints` or `container`.

Reply:
👍